src/test/java/io/cdap/plugin/gcp/bigquery/sqlengine/BigQueryReadDatasetTest.java

```diff
@@ -0,0 +1,130 @@
+                                         datasetProject, dataset, table, filter);
+    Assert.assertEquals(expectedQuery, generatedQuery);
+
+    //Without Filter
+    generatedQuery = bigQueryReadDataset.getQueryBuilder(sourceTable,
+                                                         sourceTableId,
+                                                         destTableId,
+                                                         fieldList,
+                                                         null, null, null)
+      .build().getQuery();
+
+    expectedQuery = String.format("SELECT %s FROM `%s.%s.%s`", String.join(",", fieldList),
+                                  datasetProject, dataset, table);
+    Assert.assertEquals(expectedQuery, generatedQuery);
+  }
+
+  @Test
+  public void testGenerateQuery() {
+    TableId sourceTableId = TableId.of(datasetProject, dataset, table);
+    TableId destTableId = TableId.of(datasetProject, dataset, destTable);
+
+    Mockito.when(tableDefinition.getType()).thenReturn(TableDefinition.Type.TABLE);
+    Mockito.when(sourceTable.getDefinition()).thenReturn(tableDefinition);
+    Mockito.when(sqlEngineConfig.getJobPriority()).thenReturn(QueryJobConfiguration.Priority.BATCH);
+
+    //When Table is NOT PARTITIONED
+    String generatedQuery = bigQueryReadDataset.getQueryBuilder(sourceTable,
+                                                                sourceTableId,
+                                                                destTableId,
+                                                                fieldList,
+                                                                filter,
+                                                                "2000-01-01", "2000-01-01")
+      .build().getQuery();
+
+    String expectedQuery = String.format("SELECT %s FROM `%s.%s.%s` WHERE %s", String.join(",", fieldList),
+                                         datasetProject, dataset, table, filter);
+    Assert.assertEquals(expectedQuery, generatedQuery);
+
+    //When Table is PARTITIONED
+    Mockito.when(tableDefinition.getTimePartitioning()).thenReturn(timePartitioning);
+
+    generatedQuery = bigQueryReadDataset.getQueryBuilder(sourceTable,
+                                                         sourceTableId,
+                                                         destTableId,
+                                                         fieldList,
+                                                         filter,
+                                                         "2000-01-01", "2000-01-01")
+      .build().getQuery();
+
+    String partitionFilter = "TIMESTAMP(`_PARTITIONTIME`) >= TIMESTAMP(\"2000-01-01\") and " +
+      "TIMESTAMP(`_PARTITIONTIME`) < TIMESTAMP(\"2000-01-01\")";
+    expectedQuery = String.format("SELECT %s FROM `%s.%s.%s` WHERE %s and (%s)", String.join(",", fieldList),
```

Comment:
nit: The expected query should be a fully defined string, not something that is built using code. This ensures that a bug doesn't also propagate to the test itself.

Reply:
You can use the debugger to print the `expectedQuery` value and replace the string here.